feature/dynamodb/expression/condition.go

```diff
@@ -1387,8 +1395,8 @@ func Contains(nameBuilder NameBuilder, substr string) ConditionBuilder {
 //	expression.Name("InviteList").Contains("Ben")
 //	// Let :ben be an ExpressionAttributeValue representing the value "Ben"
 //	"contains (InviteList, :ben)"
-func (nb NameBuilder) Contains(substr string) ConditionBuilder {
-	return Contains(nb, substr)
+func (nb NameBuilder) Contains(val interface{}) ConditionBuilder {
```

Comment:
Do we want to open up the type completely like this? versus
```
func ContainsInt(...)
func ContainsFloat(...)
func ContainsBool(...)
```

I don't fully understand the capabilities of the `contains` dynamodb expression (does it work with maps, sublists, etc.)

Reply:
![image](https://github.com/aws/aws-sdk-go-v2/assets/6923250/c769800e-52b0-4a76-a102-c8e4d217050d)

The only snag with current implementation would be if the user provides a type that is different than the type of the list. (ie: list of strings and providing number) but I would imagine this would just fail the dynamo filter expression case. 

Only reason I opted for opening it up is that the value is cast back to interface in `ValueBuilder`. I'm happy to implement the above methods if that is desired implementation. This would leave some ambiguity around the default `Contains()` handling both substring in string and string is list of strings. (user would have no way to differentiate, even though would yield proper results)